source/Ship.cpp

```diff
@@ -1439,18 +1439,18 @@ int Ship::Scan()
 	if(government->IsPlayer() || (target->GetGovernment()->IsPlayer() && activeScanning))
 		Audio::Play(Audio::Get("scan"), Position());
 	
-	if(startedScanning && government->IsPlayer())
+	if(startedScanning && government->IsPlayer() && !target->GetGovernment()->IsPlayer())
```

Comment:
I'd argue that if you try to scan one of your own ships (for whatever reason), you should still get a confirmation of that via a message.

Escorts will never scan anything, so if a player ship is starting a scan, that ship is guaranteed to be the flagship.